homeassistant/core.py

```diff
@@ -1940,10 +1939,7 @@ def _update(
         if elevation is not None:
             self.elevation = elevation
         if unit_system is not None:
-            if unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
-                self.units = IMPERIAL_SYSTEM
-            else:
-                self.units = METRIC_SYSTEM
+            self.units = get_unit_system(unit_system)
```

Comment:
Here there is a slight change of logic. The previous implementation defaulted to METRIC if it was an unknown value.
The new implementation raises a `ValueError`.

Is this correct?

Reply:
Let's not change the logic

Reply:
I have adjusted this to use try...except to the initial logic is maintained.